src/storage/src/table/state_table.rs

```diff
@@ -203,33 +203,46 @@ impl<S: StateStore> StateTable<S> {
     /// This function scans rows from the relational table with specific `pk_prefix`.
     pub async fn iter_with_pk_prefix(
         &self,
-        pk_prefix: Row,
+        pk_prefix: Option<&Row>,
```

Comment:
When will the `pk_prefix` be `None`? Can we just reuse `iter`? 😁

Reply:
Yes it's exactly self.iter(), I just copy the code. But seems like we can not simply write self.iter() due to some compiler problems (Checked this with @skyzh before). 

BTW `StateTableRowIter` seems needs some refactor, it may need to expose lifetime. Or we should just simply remove it.